ldes-server-port-fetch-rest/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/rest/treenode/TreeNodeFactoryImpl.java

```diff
@@ -29,7 +34,13 @@ public TreeNode getTreeNode(LdesFragmentIdentifier treeNodeId, String hostName,
 		Fragment fragment = fragmentRepository.retrieveFragment(treeNodeId)
 				.orElseThrow(
 						() -> new MissingFragmentException(extendedTreeNodeId));
-		List<Member> members = memberRepository.getMembersByReference(treeNodeId.asString()).toList();
+
+		List<Member> members = allocationRepository.findMembersForFragment(treeNodeId.asString())
+				.stream()
+				.map(memberRepository::findById)
```

Comment:
Calling the memberRepository directly kind of goes against the modulith architecture as well.. As the api of the ingest module are the interfaces in be.vlaanderen.informatievlaanderen.ldes.server.ingest. Java does not allow it but everything inside the packages should be 'hidden' from other modules (spring-modulith implements it this way). I feel like the repository should stay property of the ingest module so that this can easily change without having to worry about other modules being impacted.